pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/MessageDeduplication.java

```diff
@@ -323,26 +323,29 @@ public MessageDupStatus isDuplicate(PublishContext publishContext, ByteBuf heade
 
         String producerName = publishContext.getProducerName();
         long sequenceId = publishContext.getSequenceId();
+        headersAndPayload.markReaderIndex();
+        MessageMetadata msgMetadata = Commands.parseMessageMetadata(headersAndPayload);
```

Comment:
This will potentially increase the complexity of the dedup process because now we need to call `Commands.parseMessageMetadata(headersAndPayload);` every time. 

Can we use PublishContext.isChunked and parse the metadata only when msg is chunked? (and is chunkId only available in metadata?)

Reply:
And also please discuss this complexity increase in the PIP too.